presto-product-tests/bin/run_on_docker.sh

```diff
@@ -143,38 +131,12 @@ fi
 # catch terminate signals
 trap terminate INT TERM EXIT
 
-if [[ "$ENVIRONMENT" == "singlenode-sqlserver" ]]; then
-  EXTERNAL_SERVICES="hadoop-master sqlserver"
-elif [[ "$ENVIRONMENT" == "singlenode-ldap" ]]; then
-  EXTERNAL_SERVICES="hadoop-master ldapserver"
-elif [[ "$ENVIRONMENT" == "singlenode-mysql" ]]; then
-  EXTERNAL_SERVICES="hadoop-master mysql"
-elif [[ "$ENVIRONMENT" == "singlenode-postgresql" ]]; then
-  EXTERNAL_SERVICES="hadoop-master postgres"
-elif [[ "$ENVIRONMENT" == "singlenode-cassandra" ]]; then
-  EXTERNAL_SERVICES="hadoop-master cassandra"
-elif [[ "$ENVIRONMENT" == "singlenode-kafka" ]]; then
-  EXTERNAL_SERVICES="hadoop-master kafka"
-else
-  EXTERNAL_SERVICES="hadoop-master"
-fi
-
 # display how test environment is configured
 environment_compose config
 
-environment_compose up -d ${EXTERNAL_SERVICES}
-
-# start docker logs for the external services
-environment_compose logs --no-color -f ${EXTERNAL_SERVICES} &
-
-HADOOP_LOGS_PID=$!
-
-# start presto containers
-environment_compose up -d ${PRESTO_SERVICES}
-
-# start docker logs for presto containers
-environment_compose logs --no-color -f ${PRESTO_SERVICES} &
-PRESTO_LOGS_PID=$!
+environment_compose up -d
```

Comment:
Just curious, how does this know what needs to be started for the given "$ENVIRONMENT" setting?

Reply:
It starts up everything. The environment's docker-compose's files define what's actually inside.
By default there is only hadoop-master and presto-master (by means of `presto-product-tests/conf/docker/common/standard.yml`).
An enviornment can add more services. Or it can not depend on `standard.yml`, so that it doesn't have to have hadoop-master.